roomserver/internal/input/input_events.go

```diff
@@ -402,6 +420,94 @@ func (r *Inputer) processRoomEvent(
 	return nil
 }
 
+// processStateBefore works out what the state is before the event and
+// then checks the event auths against the state at the time. It also
+// tries to determine what the history visibility was of the event at
+// the time, so that it can be sent in the output event to downstream
+// components.
+// nolint:nakedret
```

Comment:
I'm with the linter on this one. This function is huge, and having naked returns make it hard to figure out _what_ is returned.